tests/ui/panic_in_result.rs

```diff
@@ -0,0 +1,52 @@
+#![warn(clippy::panic_in_result)]
+
+struct A;
+
+impl A {
+    fn result_with_panic() -> Result<bool, String> // should emit lint
+    {
+        panic!("error");
+    }
+
+    fn result_with_unimplemented() -> Result<bool, String> // should emit lint
+    {
+        unimplemented!();
+    }
+
+    fn result_with_unreachable() -> Result<bool, String> // should emit lint
+    {
+        unreachable!();
+    }
+
+    fn result_with_todo() -> Result<bool, String> // should emit lint
```

Comment:
The formatting looks strange here, our rustfmt config would probably change this to a more condensed style.

I'd also like to have 1-2 functions that are not methods. :)

Reply:
ran cargo dev fmt, is there a different command for rustfmt?